### PR TITLE
Remove in cluster tests

### PR DESCRIPTION
**Pull request by fgimenez**
These changes address part of the excellent feedback given so far, with them the sonobuoy tests are removed (a separate conformance project is on its way), only out of cluster tests are taken into account. I think it makes sense to not mix the conformance test execution with potentially shorter CI cycles. This by itself simplifies the test execution greatly, there's no need to build a Docker image for the tests, they can be regular go tests guarded by a build tag and, because of that, the dependencies of the e2e test don't affect unit tests execution times. 

The `e2e/project.yaml` is also simpler now, only configuration options for setting environment variables for the execution (with interpolation from the current environment available). 

What is kept from the previous implementation is the execution of the tests in a container, imo this is essential for having consistent results in different execution environments.

As before, after starting minikube with:
```
$ minikube start --extra-config=apiserver.Authorization.Mode=RBAC
```
the lifecycle will look like:
```
$ e2e-harness init # only if ./e2e doesn't exists
[ hack on ./e2e directory... ]
$ e2e-harness setup --remote=false
$ e2e-harness test # repeat as needed
$ e2e-harness teardown
```

### CODE REVIEW

**Comment by MarcelMue** [comment]
I like this change, thanks for the explanation. Are we going to do sonobuoy conformance elsewhere now?

**Comment by fgimenez** [comment]
@MarcelMue sure, I was planning to move all the code in this project that allowed us to configure, install, run and retrieve results from sonobuoy to a different repo, not yet in place though.

File: README.md [hunk 1]
```diff
@@ -138,90 +85,71 @@ It takes care of:
...
 
-First, the out of cluster tests are executed, if any, defined as
-regular `step` entries under the `outOfClusterTests` key. Then the in cluster tests,
-if they are enabled in `project.yaml`. You should enabled them only if they are
-present, if not enabling them causes an error, see above for a descrition about
-how to write this kind of tests.
+First, the binary and docker image for the project are built. Then the tests
+are compiled, not executed directly. Finally, that binary is executed in the
+test container with an environment defined by the environment variables set
+in the `project.yam` file.
```
Comment (MarcelMue) [inline README.md]: Typo: `project.yaml`

File: README.md [hunk 2]
```diff
@@ -138,90 +85,71 @@ It takes care of:
...
 
-It includes:
-  - Run specific teardown steps: this are defined in the project file
-  `e2e/project.yaml` under the `teardown` key. They are common steps (see their
-  description above) and can do things like removing the chart under test,
-  tearing down required external resources, etc.
-  - Run commomn tear down steps: these differ depending on the mode of operation,
-  for remote, ephemeral clusters they are just deleted, for local clusters tiller
-  and all the required RBAC setup is removed.
+It consists of run commomn tear down steps: these differ depending on the mode
```
Comment (MarcelMue) [inline README.md]: Better english "It consists of commonly run tear down steps" - I am not exactly sure what you mean here.

File: README.md [hunk 3]
```diff
@@ -138,90 +85,71 @@ It takes care of:
...
 
-### In cluster tests
+## Writing tests
 
-The in cluster tests execution are controlled by the `inClusterTests` entry in the project file:
+e2e tests are executed from the test container and are regular go test, for writing them
+just keep in mind these considerations:
 
+* The tests will be executed from the test container, this is it's [Dockerfile](https://github.com/giantswarm/e2e-harness/blob/master/Dockerfile).
+* All the go files should be guarded by a `e2e` build tag, being their first line:
```
Comment (MarcelMue) [inline README.md]: Just for my understanding: This means, that the very first line in my test file should be `// build + e2e` right?

File: e2e/client.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// +build e2e
```
Comment (MarcelMue) [inline e2e/client.go]: Here you start with `+` in the readme it's `build + e2e`

File: README.md (hunk 3 above)
Comment (fgimenez) [reply README.md]: Yep, the first line should be `// +build e2e` (wrong here, right below)

File: e2e/client.go (hunk 4 above)
Comment (fgimenez) [reply e2e/client.go]: Yes, this one is ok (== verified on ci :)

File: README.md (hunk 1 above)
Comment (fgimenez) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (fgimenez) [reply README.md]: Yep, finally `consists of running common steps...`

File: e2e/client.go (hunk 4 above)
Comment (kopiczko) [reply e2e/client.go]: FYI it's k8srequired now.

**Pull request merged**